### PR TITLE
[Owners] Adding licenses to build artifacts

### PR DESCRIPTION
**Pull request by SparkingSpork**
### What does this Pull Request accomplish?
Adds LICENSE and ThirdPartyNotices.txt to build artifacts

### Why should this Pull Request be merged?
Makes packaging releases easier and ensures people who download our artifacts get a copy of the licenses.

### What testing has been done?
Testing will be a run on main once this PR is approved.

### CODE REVIEW

**Review by amstewart — APPROVED** [review]

File: .github/workflows/build_nilrt.yml [hunk 1]
```diff
@@ -134,8 +134,8 @@ jobs:
           message(FATAL_ERROR "Build failed")
         endif()
 
-    - name: Tar Build Files
-      if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/releases') }}
+    - name: Tar Development Build Files
+      if: ${{ github.ref == 'refs/heads/main' }}
       run: >-
         tar -cvf NILinuxRealTime-x86_64.tar
```
Comment (kreimnitz) [inline .github/workflows/build_nilrt.yml:140]: If we make these -cvzf will Github still automatically zip them?

File: .github/workflows/build_nilrt.yml [hunk 2]
```diff
@@ -147,6 +147,20 @@ jobs:
...
         test_mutual_tls_config.json
         UnitTestsRunner
+        -C ${GITHUB_WORKSPACE}
+        LICENSE
+        ThirdPartyNotices.txt
+
+    - name: Tar Release Build Files
+      if: ${{ startsWith(github.ref, 'refs/heads/releases') }}
+      run: >-
+        tar -cvf NILinuxRealTime-x86_64.tar
```
Comment (kreimnitz) [inline .github/workflows/build_nilrt.yml:157]: Might be nice to rename the artifacts to exactly match the release names:

Linux Desktop: 
ni-grpc-device-server-linux-glibc2_23-x64.tar.gz  (created with "tar -cvzf")

LinuxRT: 
ni-grpc-device-server-ni-linux-rt-x64.tar.gz   (created with "tar -cvzf")

Windows: 
ni-grpc-device-server-windows-x64.zip

File: .github/workflows/build_nilrt.yml (hunk 1 above)
Comment (SparkingSpork) [reply .github/workflows/build_nilrt.yml:140]: Yes. All artifact downloads are zipped, even if you've already zipped them once. Just an annoying thing about how GitHub hosts them.

**Review by kreimnitz — COMMENTED** [review]
Just added some minor naming comments, thanks for taking care of this!

File: .github/workflows/build_nilrt.yml (hunk 1 above)
Comment (kreimnitz) [reply .github/workflows/build_nilrt.yml:140]: Probably still worth using -cvzf and ending files with .tar.gz I guess. We'll still have to unzip it to get down to the .tar.gz, but that's not too bad

File: .github/workflows/build_nilrt.yml (hunk 2 above)
Comment (SparkingSpork) [reply .github/workflows/build_nilrt.yml]: Done

File: .github/workflows/build_nilrt.yml (hunk 1 above)
Comment (SparkingSpork) [reply .github/workflows/build_nilrt.yml]: I've gone back and forth on this a couple of times. I don't really like having the extra layer of compression (Linux can handle zip files without issue I believe) but after thinking about it more, the extra compression to save space on our artifact storage is probably worthwhile even if we currently have unlimited space for the public repo.

Fixed.

**Review by kreimnitz — APPROVED** [review]

File: .github/workflows/build_nilrt.yml (hunk 1 above)
Comment (jonlugoatx) [reply .github/workflows/build_nilrt.yml]: The more important aspect of this is the ease of creating the release. I don't want to have to unzip the files then tar.gz them up. That adds an extra source of possible user error. If we zip up the compressed .tar.gz, then unzipping will give you the right thing to upload to the release.

**Review by jonlugoatx — APPROVED** [review]

**Pull request merged**